### ISSUE TITLE
[APP] Zuora

### ISSUE THREAD

**Issue by raphaelvarieras** [PipedreamHQ/pipedream#2639]
Zuora

https://www.zuora.com/developer/api-reference/

Action or trigger would be new ZOQL query record or full query results, as well as particular events like invoice generated, subscription created, amended, etc.

**Comment by dannyroosevelt** [comment]
Thanks for submitting the request, @raphaelvarieras. We've started the integration process but are waiting on a response from Zuora so we can create an account and create an OAuth app.

**Comment by JustinMason** [comment]
Do you have a support ticket or contact at Zuora you have been working with?  We are Zuora customers and I'd like to help get you introduced.  This integration would encourage internal use of this project.

**Comment by dannyroosevelt** [comment]
@JustinMason that'd be great if you could connect us directly. I had a thread back in April that didn't really go anywhere. Feel free to cc integrations [at] pipedream.com

**Comment by dylburger** [comment]
Without more information, I'm going to close this issue. Feel free to reopen and add more details, if you'd like.

**Comment by JustinMason** [comment]
Thanks a lot, I was able to test the basic example using oauth and product catalog and that worked.
I am curious on how events would be implemented.  Zuora has web hooks for some of their events.  They are designed to be notifications, rather than events for data integration.  But they may be useful.  We will be looking at this in the next month and I'll post here if we clarify what we may desire from Pipedream.  For larger data ETL processes Zuora provides a AQuA API that supports async bulk/batch extracts for data replication to analytic systems.  We use this for replicating data to our analytics platform.  It would be nice to have a Pipedream integration with those data extracts, in addition to the Event/Web Hooks for notifications.  
I have been impressed with the platform and community support.
